iothub/device/src/Common/Exceptions/DeviceDisabledException.cs

```diff
@@ -1,79 +0,0 @@
-// Copyright (c) Microsoft. All rights reserved.
-// Licensed under the MIT license. See LICENSE file in the project root for full license information.
-
-using System;
-using System.Runtime.Serialization;
-using Microsoft.Azure.Devices.Client.Extensions;
-
-namespace Microsoft.Azure.Devices.Client.Exceptions
-{
-    // TODO: #707 - This exception is not thrown by any protocol.
```

Comment:
Can we remove it when it's public? Also what is this todo for?

Reply:
The TODO was probably added to ensure that either the mapping is correct (what service throws vs what the sdk throws), or to remove this type.

Reply:
Ideally we shouldn't be removing any public types/ APIs, but in this case, this was unreferenced code. 
Removing such unreferenced types is going to be more of a case-by-case decision - in this case, since the SDK didn't throw this exception to the calling application, I didn't see a huge impact (a user would have had to scrub through our code to discover this exception, and add a reference to this in their application).